compiler/parser/chapel.ypp

```diff
@@ -375,12 +375,13 @@
 %left TSHIFTLEFT TSHIFTRIGHT
 %right TUPLUS TUMINUS
 %left TSTAR TDIVIDE TMOD
-%right TBNOT TBANG
+%right TBNOT TLNOT
 %left TREDUCE TSCAN TDMAPPED
 %right TEXP
 %left TCOLON
-%left TBORROWED TOWNED TUNMANAGED TSHARED
-%right TQUESTION
+%right TBORROWED TOWNED TUNMANAGED TSHARED
+%left TBANG
+%left TQUESTION
```

Comment:
Since I don't think there is an precedence that can apply between ! and ? I think we should try putting them on the same `%left TQUESTION TBANG` line.